content/linux-in-cloudnative.md

```diff
@@ -0,0 +1,73 @@
+
+What has changed, is our expectations of how they are managed once built. 
+
+
+While writing this article, an extremely interesting example came up in the form of AWS Kinesis failing and taking down us-east-1 for a fairly large amount of time. The root cause of this failure was a innocuous-seeming increase in backend capacity. This caused a over-run of the linux processes allocated in the front-end which cascaded into a widespread failure. 
+
+One main take away from this failure is that if the AWS operations team can make this mistake, anyone can. (It definitely makes me feel better about having some odd instability in our developer sandbox due to inotify-limits being met!)
+
+
+We likely don't need to know or keep track of all of these different things, in our day to day work in the cloud, but we do need to remember that Kubernetes is just a nice set of abstractions over Linux. 
+
+
+I am not claiming to be a Linux expert but have put together a list of places where I've run into issues or seen others struggle. This is by no means exhaustive, and likely doesn't even scratch the surface of where you could potentially run into "legacy" issues while running in the cloud! 
+
+
+
+# Resource Allocations and Limits
+
+Linux has a lot of resource constraints built into the OS and these can cause failures in odd and unexpected ways.
+
+As we saw in the AWS example, process limits exist!
+
+There are also file descriptor limits, inotify limits, port allocation limits, and similar. And for more high performance data centers, we should also keep in mind physical constraints as well (such as disk speed, network latency, heat dissipation, etc...)
+
+For a full list of tuning parameters on SUSE Enterprise Linux and OpenSUSE, check out https://documentation.suse.com/sles/15-SP2/single-html/SLES-tuning/#book-sle-tuning
+
+*TODO* Find more human readable list of parameters?
+
+# Build Process
+
+The processes for building applications don't really change much. Most CI/CD pipelines are really just directed graphs of fancy scripts. This means that the build tooling shouldn't change much other than the way it gets triggered.
+
+We still need to be concerned with aspects like artifact size, build time, governance, and reproducability to be able to best help the Ops team do their job! For example, knowing the dependencies that go into a build and being able to track them to the final product lets us know where potential issues might pop up before they do.
+
+
+# Security! 
+
+NOTE: I'm also not a security expert so the main point is that nothing has changed, and there's just more work to do as Kubernetes is insecure by default. 
```

Comment:
Rephrase to something like "the additional layers like container orchestration increases the attach surface."